crates/core/c8y_api/src/smartrest/operations.rs

```diff
@@ -16,6 +16,27 @@ use std::time::Duration;
 const DEFAULT_GRACEFUL_TIMEOUT: Duration = Duration::from_secs(3600);
 const DEFAULT_FORCEFUL_TIMEOUT: Duration = Duration::from_secs(60);
 
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum ResultFormat {
+    Text,
+    CSV,
+}
+
+impl<'de> Deserialize<'de> for ResultFormat {
+    fn deserialize<D>(deserializer: D) -> Result<ResultFormat, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let val: String = serde::Deserialize::deserialize(deserializer)?;
+
+        match val.as_str() {
+            "text" => Ok(ResultFormat::Text),
+            "csv" => Ok(ResultFormat::CSV),
```

Comment:
Is there an easier way to achieve this deserialization without this `Deserialize` impl? Any tricks with field attributes possible?

Reply:
You can use [`try_from` and `into` serde container attributes](https://serde.rs/container-attrs.html#try_from) to deserialize via string. It also makes the type itself serializable in other contexts.

```rust

#[derive(Debug, Clone, Eq, PartialEq, Default, Serialize, Deserialize)]
#[serde(try_from = "String", into = "String")]
pub enum ResultFormat {
    #[default]
    Text,
    CSV,
}

#[derive(Debug, thiserror::Error)]
#[error("Unknown format {0:?} - expected \"text\" or \"csv\"")]
pub struct UnknownFormatError(String);

impl TryFrom<String> for ResultFormat {
    type Error = UnknownFormatError;

    fn try_from(value: String) -> Result<Self, Self::Error> {
        match value.as_str() {
            "text" => Ok(ResultFormat::Text),
            "csv" => Ok(ResultFormat::CSV),
            _ => Err(UnknownFormatError(value)),
        }
    }
}

impl From<ResultFormat> for String {
    fn from(value: ResultFormat) -> String {
        match value {
            ResultFormat::CSV => "csv".to_string(),
            ResultFormat::Text => "text".to_string(),
        }
    }
}
```

Reply:
Those `TryFrom` impls definietly look more natural, but is pretty much the same amount of code as the existing `deserialize_with` impl. So, I'll keep the current impl for now. I was looking for something even more concise that'll do some "magic mapping" from a string to the enum variant with the same text with a single attribute. Looks like there's no much magic available :)